### ISSUE TITLE
Invoke-IcingaCheckDiskHealth Returns Inappropriate Output on Timeout

### ISSUE THREAD

**Issue by iperry-indigex** [Icinga/icinga-powershell-plugins#382]
If the disk health check times out after obtaining a value, it will return [OK] in the output, but because of the timeout, will return with status code 3.
![image](https://github.com/Icinga/icinga-powershell-plugins/assets/82535494/105b3ac2-b4f7-471c-85b2-06c37c4f9fc7)

**Comment by iperry-indigex** [comment]
Anyone wanna take a shot at this?

**Comment by LordHepipud** [comment]
Thank you for the issue. I'm not even sure on how this can happen. Is this reproducible on a constant frame or does it happen random?

It looks to me output is written to the CLI and during that time, Icinga will read everything from the buffer and kills the plugin at the same moment. `<Timeout exceeded>` is an Icinga thing. Which means, the plugin output is generated and killed before exit code 0 is reached.

**Comment by iperry-indigex** [comment]
I have only seen it happen in this one occasion when my boss found it. So in all reality, the output is correct? The plugin itself returns good information and then while cleaning up, icinga kills it for a timeout?

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by LordHepipud** [comment]
Yeah, the plugin itself reports a healthy state, but before it is exited with code 0 in this case, it is terminated by Icinga with exit code 3 and a timeout exceeded message.

Didn't even know Icinga reads plugin outputs while in a timeout scenario.

@Al2Klimov do you have an idea on how this can happen? To me it looks like the chances to win a lottery is higher than achieving this behaviour😂

**Comment by Al2Klimov** [comment]
Exactly. xD But seriously, too late by 1ms is still too late, technically. So you seem to be right: https://github.com/Icinga/icinga-powershell-plugins/issues/382#issuecomment-1925277090

**Comment by LordHepipud** [comment]
We made some smaller improvements to the disk health plugin which hopefully boosts performance enough that this won't happen in the future. Otherwise increasing the Icinga timeout or using the Icinga for Windows API might help
https://icinga.com/docs/icinga-for-windows/latest/doc/110-Installation/30-API-Check-Forwarder/
